gnat2goto/driver/ureal_to_binary.adb

```diff
@@ -0,0 +1,228 @@
+with Uintp; use Uintp;
+with Types; use Types;
+with Ada.Text_IO;
+
+package body Ureal_To_Binary is
+   package IO renames Ada.Text_IO;
+
+   procedure Get_Integer_And_Fractional_Parts (Number : Ureal;
+                                              Int_Part : out String;
+                                              Frac_Part : out String;
+                                              Int_End : out Integer;
+                                              Frac_End : out Integer)
+     with Pre => (Denominator (Number) > Uint_0);
+
+   procedure Get_Integer_And_Fractional_Parts (Number : Ureal;
+                                               Int_Part : out String;
+                                               Frac_Part : out String;
+                                               Int_End : out Integer;
+                                               Frac_End : out Integer) is
+      Num : Uint := Numerator (Number);
```

Comment:
Why is `Numerator` non-constant while `Denominator` is?

Reply:
The numerator isn't non-constant - Num is semantically more the "current remainder" of the operations below. It starts out with the numerator of the number, but it doesn't stay like that.